packages/jest-util/src/buffered_console.js

```diff
@@ -30,28 +43,110 @@ export default class BufferedConsole extends Console {
   ) {
     const call = callsites()[level != null ? level : 2];
     const origin = call.getFileName() + ':' + call.getLineNumber();
-    buffer.push({message, origin, type});
+
+    buffer.push({
+      message,
+      origin,
+      type,
+    });
+
     return buffer;
   }
 
-  debug() {
-    BufferedConsole.write(this._buffer, 'debug', format.apply(null, arguments));
+  _log(type: LogType, message: LogMessage) {
+    BufferedConsole.write(
+      this._buffer,
+      type,
+      '  '.repeat(this._groupDepth) + message,
+      3,
+    );
+  }
+
+  assert(...args: Array<any>) {
+    if (args[0]) {
+      this._log('assert', format(...args.slice(1)));
+    }
   }
 
-  log() {
-    BufferedConsole.write(this._buffer, 'log', format.apply(null, arguments));
+  count(label: string = 'default') {
+    if (!this._counters[label]) {
+      this._counters[label] = 0;
+    }
+
+    this._log('count', format(`${label}: ${++this._counters[label]}`));
+  }
+
+  countReset(label: string = 'default') {
+    this._counters[label] = 0;
   }
 
-  info() {
-    BufferedConsole.write(this._buffer, 'info', format.apply(null, arguments));
+  debug(...args: Array<mixed>) {
+    this._log('debug', format.apply(null, arguments));
```

Comment:
any reason this (and the others) are still `format.apply(null, arguments)` instead of `format(...args)`?

Reply:
We can change it to `format(...args)` everywhere, but then we'll need to change the flow notation`...args: Array<mixed>` to `...args: Array<string>`.

Reply:
That's fine, isn't it?

Reply:
If we'll decide to go with changing the signature's type notation, we'll need to also do it [in the console type definition](https://github.com/facebook/jest/blob/master/flow-typed/console.js#L22). Maybe i'm getting something wrong here, I would love to know your thoughts on this.

Reply:
thinking about it, doing `console.log({})` is perfectly valid, so `mixed` is correct

Reply:
That's true.
What do you say about `(...args: Array<any>)` and `format(...args)`?